builtins: fix out of bounds max_decimal_digits

otan opened the pull request:
max_decimal_digits when too high or too low makes strconv.FormatFloat
panic. Rectify this by constricting the bounds.

Resolves https://github.com/cockroachdb/cockroach/issues/50381.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50398)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50398#-:-MAQsfAIBp-bWNogeYwI:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/sem/builtins/geo_builtins.go, line 633 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50398#-MAQrNTM8f0bbun4-A8x:-MAQrNTM8f0bbun4-A8y:blkepp3) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a13ec0ba048873c9b47a8fe8b976145c1fe298e/pkg/sql/sem/builtins/geo_builtins.go#L633)):*
> ```Go
> 			},
> 			Info: infoBuilder{
> 				info: "Returns the WKT representation of a given Geometry. The max_decimal_digits parameter controls the maximum decimal digits to print after the `.`. Use -1 to print as many digits required to rebuild the same number.",
> ```

nit: ... as many digits as required to ...
___
*[pkg/sql/sem/builtins/geo_builtins.go, line 651 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50398#-MAQsVTr7s2UqJZzvq9-:-MAQsVTr7s2UqJZzvq90:b-rq9gf0) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f007c91e38be3fade943fa8a657b540b980fae5/pkg/sql/sem/builtins/geo_builtins.go#L651)):*
> ```Go
> 				maxDecimalDigits := int(tree.MustBeDInt(args[1]))
> 
> 				if maxDecimalDigits < -1 {
> ```

So -2 should have been allowed?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50398)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/sql/sem/builtins/geo_builtins.go, line 633 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50398#-MAQrNTM8f0bbun4-A8x:-MARQd-mBL98KF3RSg6l:b-7rl0jq) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a13ec0ba048873c9b47a8fe8b976145c1fe298e/pkg/sql/sem/builtins/geo_builtins.go#L633)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

nit: ... as many digits as required to ...
</blockquote></details>

done!
___
*[pkg/sql/sem/builtins/geo_builtins.go, line 651 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50398#-MAQsVTr7s2UqJZzvq9-:-MARQeJU8uhjwoeVc29y:b6obslo) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f007c91e38be3fade943fa8a657b540b980fae5/pkg/sql/sem/builtins/geo_builtins.go#L651)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

So -2 should have been allowed?
</blockquote></details>

yeah, i mean, it's up to us and i don't really mind :P 


<!-- Sent from Reviewable.io -->

otan committed:
builtins: fix out of bounds max_decimal_digits

max_decimal_digits when too high or too low makes strconv.FormatFloat
panic. Rectify this by constricting the bounds.

Release note: None

otan commented:
bors r=sumeerbhola
